server/graphql/v1/mutations/collectives.js

```diff
@@ -380,6 +380,38 @@ export function editCollective(_, args, req) {
 
   let collective, parentCollective;
 
+  const tiers = args.collective.tiers;
+  forEach(tiers, tier => {
+    const presets = tier.presets;
+    const amount = tier.amount;
+    const name = tier.name;
+
+    if (!tier.amountType) {
+      tier.amountType = presets ? 'FLEXIBLE' : 'FIXED';
+    }
+
+    const amountType = tier.amountType;
+    const minPreset = presets ? Math.min(...presets) : null;
+
+    if (!name || name.trim().length === 0) {
+      throw new Error('Name field is required for all tiers');
+    }
+
+    if (tier.type !== 'TICKET' && amountType === 'FIXED' && !amount) {
```

Comment:
This validations should happen at the model level, this will not cover tiers edited with `editTier` nor `editTiers` mutations.

Reply:
Good point.